public/app/features/alerting/unified/NotificationPolicies.tsx

```diff
@@ -0,0 +1,330 @@
+  MuteTimings = 'mute_timings',
+}
+
+const AmRoutes = () => {
+  const dispatch = useDispatch();
+  const styles = useStyles2(getStyles);
+
+  const [queryParams, setQueryParams] = useQueryParams();
+  const { tab } = decodeQueryParams(queryParams);
+
+  const [activeTab, setActiveTab] = useState<ActiveTab>(tab);
+  const [updatingTree, setUpdatingTree] = useState<boolean>(false);
+  const [contactPointFilter, setContactPointFilter] = useState<string | undefined>();
+  const [labelsFilter, setLabelsFilter] = useState<Label[] | undefined>();
+
+  const alertManagers = useAlertManagersByPermission('notification');
+  const [alertManagerSourceName, setAlertManagerSourceName] = useAlertManagerSourceName(alertManagers);
+
+  const amConfigs = useUnifiedAlertingSelector((state) => state.amConfigs);
+  const contactPointsState = useGetContactPointsState(alertManagerSourceName ?? '');
+
+  useEffect(() => {
+    if (alertManagerSourceName) {
+      dispatch(fetchAlertManagerConfigAction(alertManagerSourceName));
+    }
+  }, [alertManagerSourceName, dispatch]);
+
+  const {
+    result,
+    loading: resultLoading,
+    error: resultError,
+  } = (alertManagerSourceName && amConfigs[alertManagerSourceName]) || initialAsyncRequestState;
+
+  const config = result?.alertmanager_config;
+  const receivers = config?.receivers ?? [];
+
+  const rootRoute = useMemo(() => {
+    if (config?.route) {
+      return addUniqueIdentifierToRoute(config.route);
+    }
+
+    return;
+  }, [config?.route]);
+
+  // these are computed from the contactPoint and labels filter
+  const routesMatchingFilters = useMemo(() => {
+    let matchingRoutes: RouteWithID[] = [];
+    if (!rootRoute) {
+      return matchingRoutes;
+    }
+
+    const fullRoute = computeInheritedTree(rootRoute);
+
+    const routesMatchingContactPoint = contactPointFilter
+      ? findRoutesMatchingFilter(fullRoute, {
+          receiver: contactPointFilter,
+        })
+      : undefined;
+    if (routesMatchingContactPoint) {
+      matchingRoutes = routesMatchingContactPoint;
+    }
+
+    const routesMatchingLabels = labelsFilter ? findMatchingRoutes(rootRoute, labelsFilter) : undefined;
+    if (routesMatchingLabels) {
+      matchingRoutes = routesMatchingLabels;
+    }
+
+    if (routesMatchingContactPoint && routesMatchingLabels) {
+      matchingRoutes = intersectionBy(routesMatchingContactPoint, routesMatchingLabels, 'id');
+    }
+
+    return matchingRoutes;
+  }, [contactPointFilter, labelsFilter, rootRoute]);
+
+  const isProvisioned = Boolean(config?.route?.provenance);
+
+  const alertGroups = useUnifiedAlertingSelector((state) => state.amAlertGroups);
+  const fetchAlertGroups = alertGroups[alertManagerSourceName || ''] ?? initialAsyncRequestState;
+
+  function handleSave(partialRoute: Partial<FormAmRoute>) {
+    if (!rootRoute) {
+      return;
+    }
+    const newRouteTree = mergePartialAmRouteWithRouteTree(alertManagerSourceName ?? '', partialRoute, rootRoute);
```

Comment:
No clue actually, it's retrieved from several places: query param and localStorage. I think it can be empty when there's no value in the localstorage and no query param.